pkg/model/trace/combine.go

```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/grafana/tempo/pkg/tempopb"
 )
 
+// nolint: staticcheck
+// Deprecated: This is only capable of pairwise combination. It is replaced by Combiner.
```

Comment:
it seems like you've caught almost all of the uses of `CombineTraceProtos`. Any reason not to finish the job and remove this method?

If we do keep this method, should we just make it a wrapper for the Combiner so we only have the logic once?

Reply:
Was thinking it is useful to keep it for benchmark and logic comparisons, until the new Combiner has been more thoroughly tested, i.e. deprecate and delete in separate PRs.    The only remaining usage is in `func CombineForRead(obj []byte, dataEncoding string, t *tempopb.Trace) (*tempopb.Trace, error)` which only combines 2 segments so there's no benefit other than code cleanup.

Reply:
I would prefer just cleaning it up completely. I already feel like there's too many ways to combine objects and it's a bit confusing

If you feel strongly about keeping it though it's not a blocker. You've marked it deprecated which is a good indicator that no one should go near it unless they know what they're doing.

Reply:
Sounds good, removed completely.